src/session/model/Session.ts

```diff
@@ -1,109 +1,52 @@
     }
 
-    public getValue = <T = ISessionValue>(key: SessionKey): Maybe<T> => {
-        return Maybe.fromNullable(this.data[key]);
-    };
-
-    public getExtraData = (): Maybe<any> => Maybe.fromNullable(this.data[SessionKey.ExtraData]);
-
-    public saveExtraData = <T>(key: string, value: T): Session => {
-        this.dirty = true;
-        if (!this.data[SessionKey.ExtraData]) {
-            this.data[SessionKey.ExtraData] = {};
-        }
-
-        const extraData = this.data[SessionKey.ExtraData];
-
-        extraData[key] = value;
-
-        this.data[SessionKey.ExtraData] = extraData;
-
-        return this;
-    };
-
-    public verify = (): Either<Failure, VerifiedSession> => {
-        return VerifiedSession.verifySession(this);
-    };
-
-    public static createInstance = (object: any): Either<Failure, Session> => {
-        if (object) {
-            return Right(new Session(object));
+    public setExtraData<T>(key: string, value: T): void {
+        if (this.data[SessionKey.ExtraData] == null) {
+            this.data[SessionKey.ExtraData] = {}
         }
-        return Left(Failure(SessionParseError(object)));
-    };
-}
-
-export class VerifiedSession extends Session {
-
-    private constructor(session: Session) {
-        super();
-        this.data = session.data;
+        this.data[SessionKey.ExtraData][key] = value;
     }
 
-    public static verifySession(session: Session): Either<Failure, VerifiedSession> {
-
-        const signInInfo = session.data[SessionKey.SignInInfo];
+    public deleteExtraData(key: string): boolean {
+        return delete this.data[SessionKey.ExtraData][key];
+    }
 
+    public verify = (): void => {
+        const signInInfo = this.data[SessionKey.SignInInfo];
         if (!signInInfo) {
-            return Left(
-                Failure(SignInInfoMissingError)
-            );
+            throw new IncompleteSessionDataError(SessionKey.SignInInfo);
         }
 
         const accessToken = signInInfo[SignInInfoKeys.AccessToken];
-
         if (!accessToken || !accessToken[AccessTokenKeys.AccessToken]) {
-            return Left(
-                Failure(AccessTokenMissingError)
-            );
+            throw new IncompleteSessionDataError(SessionKey.SignInInfo, SignInInfoKeys.AccessToken);
         }
 
-        const expires = session.data[SessionKey.Expires];
-
+        const expires = this.data[SessionKey.Expires];
         if (!expires) {
-            return Left(
-                Failure(ExpiresMissingError)
-            );
+            throw new IncompleteSessionDataError(SessionKey.Expires);
         }
-        // This time corresponds to the time precisison given by the accounts service in seconds.
-        const dateNowMillis = Number(Date.now().toPrecision(10)) / 1000;
 
+        // This time corresponds to the time precision given by the accounts service in seconds.
+        const dateNowMillis = Number(Date.now().toPrecision(10)) / 1000;
```

Comment:
in the SessionErrors.ts, you spelled out Milliseconds in full

```suggestion
        const dateNowMilliseconds = Number(Date.now().toPrecision(10)) / 1000;
```